app/ts/utils/ensUtils.ts

```diff
@@ -246,59 +173,45 @@ export const isChildOwnershipBurned = (childInfo: DomainInfo) => {
 	})
 	return await client.waitForTransactionReceipt({ hash })
 }
 
-export const createSubDomain = async (account: AccountAddress, childInfo: DomainInfo, parentInfo: DomainInfo) => {
-	if (childInfo.registered) return undefined
-	const client = createWriteClient(account)
-	if (parentInfo.isWrapped) {
-		const [subDomainName] = childInfo.label.split('.')
-		if (subDomainName === undefined) throw new Error('subdomain missing')
-		const hash = await client.writeContract({
-			chain: mainnet,
-			account,
-			address: ENS_TOKEN_WRAPPER,
-			abi: ENS_WRAPPER_ABI, 
-			functionName: 'setSubnodeRecord',
-			args: [parentInfo.nameHash, subDomainName, parentInfo.owner, ENS_PUBLIC_RESOLVER, 0n, fuseNamesToUint(childFusesToBurn), parentInfo.expiry]
-		})
-		return await client.waitForTransactionReceipt({ hash })
-	}
-	else {
-		const hash = await client.writeContract({
-			chain: mainnet,
-			account,
-			address: ENS_REGISTRY_WITH_FALLBACK,
-			abi: ENS_REGISTRY_ABI, 
-			functionName: 'setSubnodeRecord',
-			args: [parentInfo.nameHash, childInfo.nameHash, account, ENS_PUBLIC_RESOLVER, 0n]
-		})
-		return await client.waitForTransactionReceipt({ hash })
-	}
-}
-
-export const setContentHash = async (account: AccountAddress, node: DomainInfo, contenthash: `0x${ string }`) => {
+export const callPetalLock = async (account: AccountAddress, domainInfos: DomainInfo[], contentHash: string) => {
 	const client = createWriteClient(account)
+	const petalLockAddress = `0xe49867d81af74b96c14b50881d7686a93b91efe2`
+	const subdomainRouteNames = domainInfos.map((x) => x.subDomain)
+	const labels = subdomainRouteNames.map((p) => {
+		const [label] = p.split('.')
+		if (label === undefined) throw new Error('not valid')
+		return label
+	})
+	const subdomainRouteNodes = subdomainRouteNames.map((pathPart) => namehash(pathPart))
+	const contenthash = tryEncodeContentHash(contentHash)
+	if (contenthash === undefined) throw new Error('content hash')
```

Comment:
this error should never occur as if its not validated, the button is disabled. Otherwise it could occur when user provides string that is not content hash: eg, "hello"